node/src/accountant/db_access_objects/pending_payable_dao.rs

```diff
@@ -388,25 +387,61 @@ mod tests {
 
         let result = subject.fingerprints_rowids(&[hash_1, hash_2]);
 
-        assert_eq!(result, vec![(Some(1), hash_1), (Some(2), hash_2)])
+        let first_expected_pair = (Some(1), hash_1);
+        assert!(
+            result.contains(&first_expected_pair),
+            "Returned rowid pairs should have contained {:?} but all it did is {:?}",
+            first_expected_pair,
+            result
+        );
+        let second_expected_pair = (Some(2), hash_2);
+        assert!(
+            result.contains(&second_expected_pair),
+            "Returned rowid pairs should have contained {:?} but all it did is {:?}",
+            second_expected_pair,
+            result
+        );
+        assert_eq!(result.len(), 2);
     }
 
     #[test]
-    fn fingerprints_rowids_when_nonexistent_record() {
+    fn fingerprints_rowids_when_nonexistent_records() {
         let home_dir = ensure_node_home_directory_exists(
             "pending_payable_dao",
-            "fingerprints_rowids_when_nonexistent_record",
+            "fingerprints_rowids_when_nonexistent_records",
         );
         let wrapped_conn = DbInitializerReal::default()
             .initialize(&home_dir, DbInitializationConfig::test_default())
             .unwrap();
         let subject = PendingPayableDaoReal::new(wrapped_conn);
         let hash_1 = make_tx_hash(11119);
         let hash_2 = make_tx_hash(22229);
+        let hash_3 = make_tx_hash(33339);
+        let hash_4 = make_tx_hash(44449);
+        {
+            // For more illustrative results with the hash_3 linking to rowid 2 instead of the ambiguous 1
+            subject
+                .insert_new_fingerprints(&[(hash_2, 8901234)], SystemTime::now())
+                .unwrap();
+            {
+                subject
+                    .insert_new_fingerprints(&[(hash_3, 1234567)], SystemTime::now())
+                    .unwrap()
+            }
+            subject.delete_fingerprints(&[1]).unwrap()
+        }
```

Comment:
I don't think that we need to create scope 2 times. The only use of scope is that if we want to create a variable and delete it within that scope. In our case, we don't need it.

Reply:
This is for emphasis. I could probably replace it with a comment.

Reply:
Solved